kiwi/config/functions.sh

```diff
@@ -1066,62 +491,138 @@ function baseSetRunlevel {
     esac
 }
 
+function baseSetupBuildDay {
+    local buildDay
+    buildDay="$(LC_ALL=C date -u '+%Y%m%d')"
+    echo "build_day=${buildDay}" > /build_day
+}
+
+function baseVagrantSetup {
+    # insert the default insecure ssh key from here:
+    # https://github.com/hashicorp/vagrant/blob/master/keys/vagrant.pub
+    mkdir -p /home/vagrant/.ssh/
+    echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key" > /home/vagrant/.ssh/authorized_keys
+    chmod 0600 /home/vagrant/.ssh/authorized_keys
+    chown -R vagrant:vagrant /home/vagrant/
+
+    # recommended ssh settings for vagrant boxes
+    echo "UseDNS no" >> /etc/ssh/sshd_config
+    echo "GSSAPIAuthentication no" >> /etc/ssh/sshd_config
+
+    # vagrant assumes that it can sudo without a password
+    # => add the vagrant user to the sudoers list
+    SUDOERS_LINE="vagrant ALL=(ALL) NOPASSWD: ALL"
+    if [ -d /etc/sudoers.d ]; then
+        echo "$SUDOERS_LINE" >| /etc/sudoers.d/vagrant
+        visudo -cf /etc/sudoers.d/vagrant
+        chmod 0440 /etc/sudoers.d/vagrant
+    else
+        echo "$SUDOERS_LINE" >> /etc/sudoers
+        visudo -cf /etc/sudoers
+    fi
+
+    # the default shared folder
+    mkdir -p /vagrant
+    chown -R vagrant:vagrant /vagrant
+
+    # SSH service
+    baseInsertService sshd
+}
+
+
 #======================================
-# baseCleanup
+# Common Functions (SUSE)
 #--------------------------------------
-function baseCleanup {
+function suseRemoveService {
+    # function kept for compatibility
+    baseRemoveService "$@"
+}
+
+function suseInsertService {
+    # function kept for compatibility
+    baseInsertService "$@"
+}
+
+function suseService {
+    # function kept for compatibility
+    baseService "$@"
+}
+
+function suseStripInitrd {
+    baseStripInitrd "$@"
+}
+
+function suseStripKernel {
+    baseStripKernel
+}
+
+function suseSetupProduct {
     # /.../
-    # Delete files from the system which are considered
-    # one time initialization files to be re-created on
-    # the final target system
+    # This function creates the /etc/products.d/baseproduct
+    # link pointing to the product referenced by either
+    # /etc/SuSE-brand or /etc/os-release or the latest .prod file
+    # available in /etc/products.d
     # ----
-    # systemd random seed
-    rm -f /var/lib/systemd/random-seed
+    local prod=undef
+    if [ -f /etc/SuSE-brand ];then
+        prod=$(head /etc/SuSE-brand -n 1)
+    elif [ -f /etc/os-release ];then
+        prod=$(grep "^NAME" /etc/os-release | cut -d '=' -f 2 | cut -d '"' -f 2)
```

Comment:
This should be flipped. Also, os-release can be sourced in a shell script, so you can simplify this.